conans/test/integration/command_v2/test_combined_pkglist_flows.py

```diff
@@ -0,0 +1,139 @@
+        c.run("list --graph=graph.json --graph-recipes=* --format=json")
+        pkglist = json.loads(c.stdout)["Local Cache"]
+        assert len(pkglist) == 2
+        assert len(pkglist["app/1.0"]["revisions"]["0fa1ff1b90576bb782600e56df642e19"]) == 0
+        assert len(pkglist["zlib/1.0"]["revisions"]["c570d63921c5f2070567da4bf64ff261"]) == 0
+
+
+class TestDownloadUpload:
+    @pytest.fixture()
+    def client(self):
+        c = TestClient(default_server_user=True)
+        c.save({
+            "zlib.py": GenConanfile("zlib"),
+            "zli.py": GenConanfile("zli", "1.0.0")
+        })
+        c.run("create zli.py")
+        c.run("create zlib.py --version=1.0.0 --user=user --channel=channel")
+        c.run("upload * -r=default -c")
+        c.run("remove * -c")
+        return c
+
+    @pytest.mark.parametrize("prev_list", [False, True])
+    def test_download_upload_all(self, client, prev_list):
+        # We need to be consequeent with the pattern, it is not the same defaults for
+        # download and for list
+        pattern = "zlib/*#latest:*#latest"
+        if prev_list:
+            client.run(f"list {pattern} -r=default --format=json", redirect_stdout="pkglist.json")
+            # Overwriting previous pkglist.json
+            pattern = "--list=pkglist.json"
+
+        client.run(f"download {pattern} -r=default --format=json", redirect_stdout="pkglist.json")
+        # TODO: Discuss "origin"
+        assert "Local Cache" in client.load("pkglist.json")
+        client.run("remove * -r=default -c")
+        client.run("upload --list=pkglist.json -r=default")
+        assert f"Uploading recipe 'zlib/1.0.0" in client.out
+        assert f"Uploading recipe 'zli/" not in client.out
+        assert "Uploading package 'zlib/1.0.0" in client.out
+        assert "Uploading package 'zli/" not in client.out
+
+    @pytest.mark.parametrize("prev_list", [False, True])
+    def test_download_upload_only_recipes(self, client, prev_list):
+        if prev_list:
+            pattern = "zlib/*#latest"
+            client.run(f"list {pattern} -r=default --format=json", redirect_stdout="pkglist.json")
+            # Overwriting previous pkglist.json
+            pattern = "--list=pkglist.json"
+        else:
+            pattern = "zlib/*#latest --only-recipe"
+        client.run(f"download {pattern} -r=default --format=json", redirect_stdout="pkglist.json")
+        # TODO: Discuss "origin"
+        assert "Local Cache" in client.load("pkglist.json")
+        # Download binary too! Just to make sure it is in the cache, but not uploaded
+        # because it is not in the orignal list of only recipes
+        client.run(f"download * -r=default")
+        client.run("remove * -r=default -c")
+        client.run("upload --list=pkglist.json -r=default")
+        assert f"Uploading recipe 'zlib/1.0.0" in client.out
+        assert f"Uploading recipe 'zli/" not in client.out
```

Comment:
This neatly shows the usecase, I think it was a good call to have both download & upload available at first :)